### ISSUE TITLE
Invalid TS config for `@markdoc/markdoc@0.1.7`

### ISSUE THREAD

**Issue by alunny-stripe** [markdoc/markdoc#195]
### What happened?

Importing `@markdoc/markdoc` in a fresh TypeScript projects shows multiple errors:

```
node_modules/@markdoc/markdoc/dist/index.d.ts:15:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                     
                                                                                                                                                                                                           
15 import type Token from 'markdown-it/lib/token';                                                                                                                                                         
                          ~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                          
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/index.d.ts:98:35 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                               
                                                                                                                                                                                                           
98         getAstValues(value: any): Generator<import("./src/types").AstType, void, unknown>;                                                                                                              
                                     ~~~~~~~~~                                                                                                                                                             
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/ast/index.d.ts:11:31 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                       
                                                                                                                                                                                                           
11     getAstValues(value: any): Generator<AstType, void, unknown>;                                                                                                                                        
                                 ~~~~~~~~~                                                                                                                                                                 
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/ast/node.d.ts:14:13 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                        
                                                                                                                                                                                                           
14     walk(): Generator<Node, void, unknown>;                                                                                                                                                             
               ~~~~~~~~~                                                                                                                                                                                   
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/renderers/react/react.d.ts:1:57 - error TS2307: Cannot find module 'react' or its corresponding type declarations.                                                  
                                                                                                                                                                                                           
1 import type { createElement, Fragment, ReactNode } from 'react';                                                                                                                                         
                                                          ~~~~~~~                                                                                                                                          
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib' or its corresponding type declarations.                                              
                                                                                                                                                                                                           
1 import MarkdownIt from 'markdown-it/lib';                                                                                                                                                                
                         ~~~~~~~~~~~~~~~~~                                                                                                                                                                 
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:2:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                        
                                                                                                                                                                                                           
2 import type Token from 'markdown-it/lib/token';                                                                                                                                                          
                         ~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                           
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/utils.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                  
                                                                                                     
1 import type Token from 'markdown-it/lib/token';                                                    
                         ~~~~~~~~~~~~~~~~~~~~~~~                                                     
```

Some of these errors are due to missing dependencies `@types/react` and `@types/markdown-it`

### To reproduce

1. Download or clone https://gist.github.com/alunny-stripe/ed7e03f7a6fd712a8ea93eee36af8dbb
2. Run `npm install`
3. Run `npm run types`

### Version

0.1.7

### Additional context

I ran into slightly different errors on my project, which is using TypeScript 4.5.4 and pnpm. Here are the TS@4.5.4 errors:

```
node_modules/@markdoc/markdoc/dist/index.d.ts:15:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                     

15 import type Token from 'markdown-it/lib/token';
                          ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/index.d.ts:98:35 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

98         getAstValues(value: any): Generator<import("./src/types").AstType, void, unknown>;
                                     ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/index.d.ts:111:48 - error TS2694: Namespace '"/Users/alunny/dev/markdoc-ts/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Variable'.

111         Variable: typeof import("./src/types").Variable;
                                                   ~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/ast/index.d.ts:11:31 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

11     getAstValues(value: any): Generator<AstType, void, unknown>;
                                 ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/src/ast/node.d.ts:14:13 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

14     walk(): Generator<Node, void, unknown>;
               ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/src/renderers/react/react.d.ts:1:57 - error TS2307: Cannot find module 'react' or its corresponding type declarations.

1 import type { createElement, Fragment, ReactNode } from 'react';
                                                          ~~~~~~~

node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib' or its corresponding type declarations.

1 import MarkdownIt from 'markdown-it/lib';
                         ~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:2:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.

2 import type Token from 'markdown-it/lib/token';
                         ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/utils.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.

1 import type Token from 'markdown-it/lib/token';
```

**Comment by mfix-stripe** [comment]
PTAL @matv-stripe @rpaul-stripe

**Comment by rlpiotrowski** [comment]
Any updates? I've run into similar issue with latest angular project.

**Comment by mfix-stripe** [comment]
Would be awesome if you could take a quick peak at this @benbayard-stripe

**Comment by alunny-stripe** [comment]
Thanks @mfix-stripe! With the changes in #197, this is the minimal `tsconfig.json` I need to get things passing in TS 4.8.3:

```json
{
  "compilerOptions": {
    "esModuleInterop": true,
    "moduleResolution": "node",
    "target": "esnext"
  }
}
```

As Ben says, it would be great to add that to the README 😅 

On TS 4.5 (which we use internally) and up to 4.7, I still get the following error:
```
node_modules/@markdoc/markdoc/dist/index.d.ts:111:48 - error TS2694: Namespace '"/Users/alunny/dev/markdoc-ts/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Variable'.

111         Variable: typeof import("./src/types").Variable;                                              ~~~~~~~~
```

I wonder if there was a change in how `esModuleInterop` in TS 4.8 🤷

**Comment by MitchellCash** [comment]
@mfix-stripe is it possible to push out a bug fix release now #197 is merged? Even if it means just cherry-picking those specific commits into a release

**Comment by MitchellCash** [comment]
Sorry to pile on a closed issue, but I am also seeing a similar issue to what @alunny-stripe raised in that when I use a TypeScript version lower than 4.8 It will throw `Variable: typeof import("./src/types").Variable;`.

TS 4.8 doesn't seem to do this for me, but TS 4.8 is only a month old and I can't upgrade my project to that version yet.
